aws-s3-transfer-manager/src/types.rs

```diff
@@ -20,17 +22,59 @@ pub enum PartSize {
     Target(u64),
 }
 
-/// The concurrency settings to use for a single upload or download request.
+/// The concurrency mode the client should use for executing requests.
 #[derive(Debug, Clone, Default)]
-pub enum ConcurrencySetting {
-    /// Automatically configure an optimal concurrency setting based on the execution environment.
+pub enum ConcurrencyMode {
+    /// Automatically configured concurrency based on the execution environment.
     #[default]
     Auto,
 
-    /// Explicitly configured concurrency setting.
+    /// Explicitly configured throughput setting the client should aim for.
+    ///
+    /// In this mode, concurrency is limited by attempting to hit a throughput target.
+    TargetThroughput(TargetThroughput),
+
+    /// Explicit concurrency control
     Explicit(usize),
 }
 
+/// Throughput target(s)
+#[derive(Debug, Clone)]
+pub struct TargetThroughput {
+    upload: Throughput,
+    download: Throughput,
+}
+
+impl TargetThroughput {
+    /// Create a new target throughput in Gigabits/sec that is
+    /// the same for both uploads and downloads.
+    pub fn new_gigabits_per_sec(gbps: u64) -> Self {
+        let target = Throughput::new_bytes_per_sec(gbps * ByteUnit::Gigabit.as_bytes_u64());
+        Self::new(target, target)
+    }
+
+    // TODO: we don't actually support limiting throughput by upload/download independently (yet), it will require updates to scheduler.
+
+    /// Create a new target throughput using the given upload and download
+    /// throughputs.
+    fn new(upload_target: Throughput, download_target: Throughput) -> Self {
```

Comment:
trivial: when separate speeds are given for download and upload, download is usually cited first